### PR TITLE
Fixed button wrapping behavior on mobile

### LINKED ISSUE #918: Next and previous buttons wrap and stack on mobile
## 🚀 Feature

Changing the next and previous button styles so that they don't stack and wrap on smaller screens (either on desktop or mobile). 

img_7537

## Have you read the [Contributing Guidelines on issues](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Motivation

The behavior happens when the text inside the buttons are long. This is especially noticeable on mobile, since it's easier to click on the wrong thing.

## Pitch

Modify button styles for mobile screen widths.

- Option 1: set both buttons to equal width on smaller screens, forcing the lengthy text to wrap within the button itself
- Option 2: set the next button to `float: left` on smaller screens and add `padding: 10px` so that there's consistency and clear separation between the two buttons

option-comparison

### PR DESCRIPTION
**Pull request by notjaril**
## Motivation

Fixes #918 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

1) Visit any doc page where lengthy text on the previous and next buttons caused the buttons to wrap and stack with no space in between (eg https://docusaurus.io/docs/en/navigation)
2) Check that it no longer stacks when the window is <735px. There should be ample space in between and the both buttons should float left.

It should look like this:
screen shot 2018-08-27 at 5 23 31 pm

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

### CODE REVIEW

File: lib/static/css/main.css [hunk 1]
```diff
@@ -1545,8 +1545,28 @@ input::placeholder {
 }
 
 @media only screen and (max-width: 735px) {
-  .docs-prevnext {
```
Comment (endiliey) [inline lib/static/css/main.css]: Deleting this means the class here is useless
https://github.com/notjaril/Docusaurus/blob/f9151ca17ac424d0102d65faedef78db15ef4cbd/lib/core/DocsLayout.js#L90

```js
<div className="docs-prevnext">
```

File: lib/core/DocsLayout.js [hunk 2]
```diff
@@ -97,16 +113,15 @@ class DocsLayout extends React.Component {
...
-                        'localized-strings'
-                      ].next ||
-                      'Next'
-                    : metadata.next_title || 'Next'}{' '}
-                  →
+                  <span
+                    className={
+                      nextTitle.match(/[a-z][A-Z]/)
+                        ? 'function-name-prevnext'
+                        : 'text-next'
```
Comment (endiliey) [inline lib/core/DocsLayout.js]: I couldn't find any `text-next` css class on the new `main.css`.

File: lib/static/css/main.css (hunk 1 above)
Comment (notjaril) [reply lib/static/css/main.css]: The docs-prevnext class is still styled with: 

```css
/* Start of Docs Navigation */
.docs-prevnext {
  margin: 20px 0;
}
```

The intent with removing the doc-prevnext styling within the media query was so the buttons' top and bottom margins were displayed properly, where it wasn't being display properly with a fixed doc-prevnext height. 

docs-prevnext

Does this make sense? Let me know if I missed something.

Comment (endiliey) [reply lib/static/css/main.css]: This is correct. I missed that :+1:

File: lib/core/DocsLayout.js (hunk 2 above)
Comment (endiliey) [reply lib/core/DocsLayout.js]: Seems that is the case. Just checking .. It's easy to miss css selector like this :smile:

**Comment by notjaril** [comment]
I like the 'before' look as well for shorter text. 

Ideally, it would keep both inline until such time that one of the buttons wrap, at which point if floats both buttons left. The problem is I don't think there's a way in CSS to toggle the styling based on whether or not the elements are wrapped. The 'after' styling just toggles automatically at screen width < 735px, floating both buttons left.

If you feel strongly about keeping both buttons inline, we could do Option 1 instead of Option 2. Then both buttons would have equal widths, and any lengthy text would wrap.
option-comparison

It's possible I'm missing something - if there's a CSS trick out there that could do that, let me know.

**Comment by notjaril** [comment]
Fixed the conflicts - let me know if this does the trick.

File: lib/core/DocsLayout.js [hunk 3]
```diff
@@ -46,6 +46,23 @@ class DocsLayout extends React.Component {
     const title =
       idx(i18n, ['localized-strings', 'docs', id, 'title']) || defaultTitle;
     const hasOnPageNav = this.props.config.onPageNav === 'separate';
+
+    const previousTitle = i18n
+      ? translation[this.props.metadata.language]['localized-strings'][
```
Comment (endiliey) [inline lib/core/DocsLayout.js]: You should do it with `idx` (safe getter) now. See the upstream master

**Comment by notjaril** [comment]
That was my bad - it's now fixed and is using idx.

**Pull request merged**